vendor/github.com/xtaci/smux/session.go

```diff
@@ -281,73 +276,76 @@ func (s *Session) recvLoop() {
 	}
 }
 
-func (s *Session) keepalive() {
-	tickerPing := time.NewTicker(s.config.KeepAliveInterval)
-	tickerTimeout := time.NewTicker(s.config.KeepAliveTimeout)
-	defer tickerPing.Stop()
-	defer tickerTimeout.Stop()
+// keepAliveSend will periodically send a keepalive meesage to the remote peer.
+func (s *Session) keepAliveSend() {
+	keepAliveTimeout := time.After(s.config.KeepAliveInterval)
```

Comment:
The ticker method means that keepalive messages can get missed. If it takes more than 1 tick to perform a keepalive send, then the timing could mean that you miss the next tick as well. Switching to an absolute timeout gives better control over when we do things.